core/koin-core-coroutines/src/commonMain/kotlin/org/koin/core/module/LazyModule.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017-Present the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.koin.core.module
+
+/**
+ * Koin Lazy Module - an implementation of [Lazy]<[Module]>.
+ *
+ * Accepts a lambda that initializes a [Module] via [lazy()][lazy]
+ * using [LazyThreadSafetyMode.NONE] as thread-safety [mode][LazyThreadSafetyMode].
+ *
+ * @author Chris Paleopanos
+ * @param moduleInitializer a lambda that will be used to initialize a [Module] lazily
+ */
+@KoinDslMarker
+class LazyModule(moduleInitializer: () -> Module) : Lazy<Module> by lazy(LazyThreadSafetyMode.NONE, moduleInitializer) {
```

Comment:
Let me know if you want me to annotate this with `@KoinExperimentalAPI`. I wasn't sure if that's desirable since that's not the behavior of the current implementation.

Annotating this class with `@KoinExperimentalAPI` will raise a warning for all modules that are constructed via `lazyModule {}` DSL.

Reply:
this is already the case for LazyModule API no? 🤔

Reply:
but I could raise it in 3.6

Reply:
Currently the warning is raised on `lazyModule {}` usage. If I add the annotation here then the warning will also be raised for `lazyKoinModule` usages too in the following example - which is not the case so far because it's type is `Lazy<Module>`.

```kotlin
val lazyKoinModule = lazyModule {}
```

It makes sense to add the annotation here, but will increase the amount of warnings dramatically - up to you!

Reply:
yes that was a case in previous Koin 2.x. Let's keep out of annotation then